packages/frontend/src/pages/campaigns/filters/checkboxes-filter/index.tsx

```diff
@@ -0,0 +1,96 @@
+import { Template } from "@carrot-kpi/sdk";
+import { Checkbox, Typography } from "@carrot-kpi/ui";
+import React, { ChangeEvent, useCallback, useEffect, useState } from "react";
+import { FiltersLoading } from "../loading";
+import {
+    CheckboxesObj,
+    generateGroupId,
+    getDefaultCheckboxes,
+    getDefaultSelected,
+} from "../utils";
+
+interface CheckboxesFilterProps {
+    items: Template[];
+    groupId: string;
+    title: string;
+    loading: boolean;
+    selected: Set<number>;
+    setSelected: (newSelected: Set<number>) => void;
+    defaultChecked: (item: Template) => boolean;
+}
+
+export const CheckboxesFilter = ({
+    items,
+    title,
+    loading,
+    selected,
+    setSelected,
+    defaultChecked,
+    groupId,
+}: CheckboxesFilterProps) => {
+    const [checkboxes, setCheckboxes] = useState<CheckboxesObj>({});
+
+    const initializeCheckboxes = useCallback(() => {
+        const hasItems = items.length !== 0;
+        const hasCheckboxes = Object.keys(checkboxes).length !== 0;
+
```

Comment:
Maybe we can rewrite this as:

```ts
if(items.length === 0) return;
if(Object.keys(checkboxes).length !== 0) return;
```

I think it's a bit more clear.

Reply:
Ok. I'll change that.

Reply:
fixed: https://github.com/carrot-kpi/v1-monorepo/pull/206/commits/c6a5a7f110066cc7fc709404d55e4b1d15027682